niworkflows/utils/bids.py

```diff
@@ -304,3 +306,26 @@ def _grp_echos(x):
         action = getattr(ses_uids, 'append' if len(bold) > 2 else 'extend')
         action(bold)
     return ses_uids
+
+
+def check_pipeline_version(cvers, data_desc):
+    """
+    Searches for existing BIDS pipeline output and compares against current pipeline version.
+    Produces a warning if there is a difference between versions.
+
+    Parameters
+    ----------
+    cvers : :obj:str
+        Current pipeline version
+    data_desc : :obj:str or Pathlike
+        Path to pipeline output's dataset_description.json
+
+    """
+    if not Path(str(data_desc)).exists():
+        return
+
+    with open(str(data_desc), encoding='utf-8') as fp:
+        desc = json.load(fp)
+    dvers = desc.get("PipelineDescription", {}).get("Version", "0+unknown")
+    if Version(cvers) != Version(dvers):
+        warnings.warn("Previous output generated by version {} found.".format(dvers))
```

Comment:
I'm inclined to think that issuing the warning is actually a responsibility of the end-user tool (and maybe redirect it to the log instead, or stop execution, etc.)

Reply:
sure, how do you feel about simply returning the string then?

Reply:
yes, that's probably how I would have imagined it.